src/main/java/org/opensearch/ad/ml/EntityColdStarter.java

```diff
@@ -694,40 +690,30 @@ public void trainModelFromExistingSamples(ModelState<EntityModel> modelState, in
     }
 
     /**
-     * Precondition: we don't have enough training data.
-     * Combine training data with existing sample data.
-     * Existing samples either predates or coincide with cold start data.  In either case,
-     * combining them without reordering based on time stamps is not ok. We might introduce
-     * anomalies in the process.
-     * An ideal solution would be to record time stamps of data points and combine existing
-     * samples and cold start samples and do interpolation afterwards. Recording time stamps
-     * requires changes across the board like bwc in checkpoints. A pragmatic solution is to use
-     * job enabled time as the end time of cold start period as it is easier to combine
-     * existing samples with cold start data. We just need to appends existing samples after
-     * cold start data as existing samples all happen after job enabled time. There might
-     * be some gaps in between the last cold start sample and the first accumulated sample.
-     * We will need to accept that precision loss in current solution.
+     * Extract training data and put them into ModelState
      *
      * @param coldstartDatapoints training data generated from cold start
      * @param modelId model Id
      * @param entityState entity State
      */
-    private void combineTrainSamples(List<double[][]> coldstartDatapoints, String modelId, ModelState<EntityModel> entityState) {
-        if (coldstartDatapoints == null || coldstartDatapoints.size() == 0) {
+    private void extractTrainSamples(List<double[][]> coldstartDatapoints, String modelId, ModelState<EntityModel> entityState) {
+        if (coldstartDatapoints == null || coldstartDatapoints.size() == 0 || entityState == null) {
             return;
         }
 
         EntityModel model = entityState.getModel();
         if (model == null) {
             model = new EntityModel(null, new ArrayDeque<>(), null);
+            entityState.setModel(model);
```

Comment:
What is the reason for setting an empty model now where previously we haven't done this.

Reply:
not setting an empty model is a bug. But because the model will not be empty in current workflow so it does not manifest itself.